### PR TITLE
Get Data from Mongo DB By ID into Backend

### PR DESCRIPTION
**Pull request by kathahuth**
- add getById for group difficulty to service
- add getById test to service-test
- add getById for group difficulty in controller
- add getById test to controller-test
- add builder to service-test and controller-test
- refactoring

### CODE REVIEW

File: backend/src/main/java/de/kathahuth/gymast/model/Group.java [hunk 1]
```diff
@@ -9,9 +10,12 @@
 @Data
 @AllArgsConstructor
 @NoArgsConstructor
+@Builder
```
Comment (lisamittmann) [inline backend/src/main/java/de/kathahuth/gymast/model/Group.java:13]: please keep in mind that `@builder` enables you to build incomplete objects which may cause problems if you rely on complete objects in your data base

File: backend/src/main/java/de/kathahuth/gymast/model/Group.java [hunk 2]
```diff
@@ -9,9 +10,12 @@
...
 @Document(collection = "groups")
 public class Group {
 
     @Id
-    private String name;
+    private String id;
```
Comment (lisamittmann) [inline backend/src/main/java/de/kathahuth/gymast/model/Group.java:18]: you can also use name as `@Id` attribute name, if you are using a name I would recommend using `name` instead

File: backend/src/test/java/de/kathahuth/gymast/controller/GroupControllerTest.java [hunk 3]
```diff
@@ -51,9 +65,34 @@ public void getAllGroups() {
...
+
+    @Test
+    @DisplayName("Get Group By Id should return GroupId")
+    public void getGroupById(){
+        //Given
+        String groupId = "Dynamic Strength";
+        groupMongoDb.save(createDynamicGroup());
+
+        //When
+        ResponseEntity<Group> response = testRestTemplate.getForEntity(getUrl() + "/" + groupId, Group.class);
```
Comment (lisamittmann) [inline backend/src/test/java/de/kathahuth/gymast/controller/GroupControllerTest.java:79]: you could also change your `getUrl()` method to end with `/` so you do not need to include it here

File: backend/src/test/java/de/kathahuth/gymast/service/GroupServiceTest.java [hunk 4]
```diff
@@ -6,31 +6,82 @@
...
+                createDynamicGroup(),
+                createJumpGroup()));
+    }
+
+    @Test
+    @DisplayName("getGroupById should return group by Id")
+    public void getGroupById(){
+
+        //Given
+        String groupId = "Dynamic Strength";
```
Comment (lisamittmann) [inline backend/src/test/java/de/kathahuth/gymast/service/GroupServiceTest.java]: if you are using descriptive Strings for your ID I would rather switch to an ID attribute without any space (IDs usually do not include spaces) e.g. `dynamicStrength` and include a description attribute you can display in your frontend

**Review by lisamittmann — APPROVED** [review]
lgtm 🧁 just a few pointers
would be great if you could include a GitHub workflow showing that your tests run successfully 🤞🏻

**Comment by kathahuth** [comment]
Added Lisas suggestions from review

**Pull request merged**